dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -779,20 +777,9 @@ public SourceFile dockerfile(@ApiParam(hidden = true) @Auth Optional<User> user,
     public String verifiedSources(@ApiParam(value = "Tool id", required = true) @PathParam("containerId") Long containerId) {
         Tool tool = toolDAO.findById(containerId);
         checkEntry(tool);
-
-        Set<String> verifiedSourcesArray = tool.getWorkflowVersions().stream()
-                .filter(Version::isVerified)
-                .map(Version::getVerifiedSource).collect(Collectors.toSet());
-
-        JSONArray jsonArray;
-        try {
-            jsonArray = new JSONArray(verifiedSourcesArray.toArray());
-        } catch (JSONException ex) {
-            throw new CustomWebApplicationException("There was an error converting the array of verified sources to a JSON array.",
-                HttpStatus.SC_INTERNAL_SERVER_ERROR);
-        }
-
-        return jsonArray.toString();
+        Set<String> verifiedSources = VerificationHelper.getVerifiedSources(tool.getWorkflowVersions());
```

Comment:
Looks like this iterates through all source files for all versions on every call